File: ReSharper.FSharp/src/Psi.FSharp.LanguageService/ZoneMarker.cs
using JetBrains.Application.BuildScript.Application.Zones;

namespace JetBrains.ReSharper.Plugins.FSharp.Psi.LanguageService
{
  [ZoneMarker]
  public class ZoneMarker : IRequire<ILanguageFSharpZone>
  {
  }
}

File: ReSharper.FSharp/src/FSharp.Psi.Services/src/ZoneMarker.fs
namespace JetBrains.ReSharper.Plugins.FSharp.Psi

open JetBrains.Application.BuildScript.Application.Zones
open JetBrains.DocumentModel
open JetBrains.ReSharper.Plugins.FSharp
open JetBrains.ReSharper.Resources.Shell

[<ZoneMarker>]
type ZoneMarker() =
    interface IRequire<IDocumentModelZone>
    interface IRequire<ILanguageFSharpZone>
    interface IRequire<PsiFeaturesImplZone>
